script4.js

```diff
@@ -0,0 +1,115 @@
+//1 While loop//
+
+const person = { name: "Jane", age: "23", city: "LA" };
+
+function printObj(person) {
+  let i = 0;
+
+  let keys = Object.keys(person);
+  let values = Object.values(person);
+  let count = keys.length; // or va
+
+  while (i < count) {
+    console.log(keys[i] + ": " + values[i]);
+    i++;
+  }
+}
+printObj(person);
+
+//2 Object search//
+const person2 = { name: "John", age: "24", city: "NY" };
+
+function searchObj(obj, key) {
+  for (let k in obj) {
+    if (k === key) {
+      return true;
+    }
+  }
+  return false;
+}
+
+console.log(searchObj(person2, "age"));
+console.log(searchObj(person2, "citya"));
+console.log(searchObj(person, "city"));
+console.log(searchObj(person, "color"));
+
+// 3. Combine propreties//
+let helloWorld = { prop1: "Hello", prop2: "World", prop3: "!" };
+
+function combineObject(objct) {
+  let objKeys = Object.keys(objct);
+  let result = "";
+  let i = 0;
+
+  do {
+    let currentKey = objKeys[i];
+    let currentValue = helloWorld[currentKey];
+    result += currentValue;
+    i++;
+  } while (i < objKeys.length);
+  return result;
+}
+
+console.log(combineObject(helloWorld));
+
+// 4 number of wowels//
+
+function countVowels(str) {
+  let count = 0;
+  const vowels = "aeiouAEIOU"; // To count both uppercase and lowercase vowels
```

Comment:
Add "y,Y" to list of vowels and you will get a right result))

Reply:
Thank you